modules/pipeline/pipengine/actionexecutor/plugins/docker/docker.go

```diff
@@ -0,0 +1,395 @@
+		}
+	}
+	jobContainer, err := d.createContainerByJob(ctx, &job)
+	if err != nil {
+		return nil, err
+	}
+	if err := d.client.ContainerStart(ctx, jobContainer.ID, dockertypes.ContainerStartOptions{}); err != nil {
+		return nil, errors.Errorf("failed to start job container, err: %v", err)
+	}
+	return apistructs.Job{
+		JobFromUser: job,
+	}, nil
+}
+
+func (d *DockerJob) Update(ctx context.Context, task *spec.PipelineTask) (interface{}, error) {
+	return nil, errors.Errorf("%s not support update operation", d.Kind().String())
+}
+
+func (d *DockerJob) Cancel(ctx context.Context, task *spec.PipelineTask) (data interface{}, err error) {
+	defer d.errWrapper.WrapTaskError(&err, "cancel job", task)
+	if err := logic.ValidateAction(task); err != nil {
+		return nil, err
+	}
+	oldUUID := task.Extra.UUID
+	task.Extra.UUID = task_uuid.MakeJobID(task)
+	data, err = d.Delete(ctx, task)
+	task.Extra.UUID = oldUUID
+	return data, err
+}
+
+func (d *DockerJob) Remove(ctx context.Context, task *spec.PipelineTask) (data interface{}, err error) {
+	defer d.errWrapper.WrapTaskError(&err, "remove job", task)
+	if err := logic.ValidateAction(task); err != nil {
+		return nil, err
+	}
+	task.Extra.UUID = task_uuid.MakeJobID(task)
+	return d.Delete(ctx, task)
+}
+
+func (d *DockerJob) BatchDelete(ctx context.Context, tasks []*spec.PipelineTask) (data interface{}, err error) {
+	if len(tasks) == 0 {
+		return nil, nil
+	}
+	task := tasks[0]
+	defer d.errWrapper.WrapTaskError(&err, "batch delete job", task)
+	for _, task := range tasks {
+		if len(task.Extra.UUID) <= 0 {
+			continue
+		}
+		_, err = d.Delete(ctx, task)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return nil, nil
+}
+
+func (d *DockerJob) Delete(ctx context.Context, task *spec.PipelineTask) (data interface{}, err error) {
+	job, err := logic.TransferToSchedulerJob(task)
+	if err != nil {
+		return nil, err
+	}
+	name := d.makeJobName(task.Extra.Namespace, task.Extra.UUID)
+	jobContainer, err := d.client.ContainerInspect(ctx, name)
+	if err != nil {
+		if client.IsErrNotFound(err) {
+			logrus.Warnf("%s: task not exist, taskInfo: %s", d.Kind().String(), logic.PrintTaskInfo(task))
+			return nil, nil
+		}
+		return nil, err
+	}
+	if err := d.client.ContainerRemove(ctx, jobContainer.ID, dockertypes.ContainerRemoveOptions{
+		Force: true,
+	}); err != nil {
+		return nil, err
+	}
+	for _, mountVol := range jobContainer.Mounts {
+		err = d.client.VolumeRemove(ctx, mountVol.Name, true)
+		if err != nil {
+			if !client.IsErrNotFound(err) {
+				return nil, errors.Errorf("failed to remove docker volume, name: %s", mountVol.Name)
+			}
+			logrus.Warningf("the docker job %s's volume %s in namespace %s is not found", name, mountVol.Name, job.Namespace)
+		}
+	}
+	return task.Extra.UUID, nil
+}
+
+func (d *DockerJob) Inspect(ctx context.Context, task *spec.PipelineTask) (apistructs.TaskInspect, error) {
+	return apistructs.TaskInspect{}, nil
```

Comment:
anything to inspect from `docker inspect`?

Reply:
added, use `docker inspect`